WalletWasabi.Gui/Rpc/WasabiJsonRpcService.cs

```diff
@@ -0,0 +1,208 @@
+using WalletWasabi.Blockchain.Transactions;
+using WalletWasabi.CoinJoin.Client.Clients.Queuing;
+using WalletWasabi.Helpers;
+using WalletWasabi.Models;
+
+namespace WalletWasabi.Gui.Rpc
+{
+    public partial class WasabiJsonRpcService
+    {
+        private Global Global { get; }
+
+        public WasabiJsonRpcService(Global global)
+        {
+            Global = global;
+        }
+
+        [JsonRpcMethod("listunspentcoins")]
+        public object[] GetUnspentCoinList()
+        {
+            AssertWalletIsLoaded();
+            return Global.WalletService.Coins.Where(x => x.Unspent).Select(x => new
+            {
+                txid = x.TransactionId.ToString(),
+                index = x.Index,
+                amount = x.Amount.Satoshi,
+                anonymitySet = x.AnonymitySet,
+                confirmed = x.Confirmed,
+                label = x.Label.ToString(),
+                keyPath = x.HdPubKey.FullKeyPath.ToString(),
+                address = x.HdPubKey.GetP2wpkhAddress(Global.Network).ToString()
+            }).ToArray();
+        }
+
+        [JsonRpcMethod("getwalletinfo")]
+        public object WalletInfo()
+        {
+            AssertWalletIsLoaded();
+            var km = Global.WalletService.KeyManager;
+            return new
+            {
+                walletFile = Global.WalletService.KeyManager.FilePath,
+                extendedAccountPublicKey = km.ExtPubKey.ToString(Global.Network),
+                extendedAccountZpub = km.ExtPubKey.ToZpub(Global.Network),
+                accountKeyPath = $"m/{km.AccountKeyPath.ToString()}",
+                masterKeyFingerprint = km.MasterFingerprint?.ToString() ?? "",
+                balance = Global.WalletService.Coins
+                            .Where(c => c.Unspent && !c.SpentAccordingToBackend)
+                            .Sum(c => c.Amount.Satoshi)
+            };
+        }
+
+        [JsonRpcMethod("getnewaddress")]
+        public object GenerateReceiveAddress(string label)
+        {
+            AssertWalletIsLoaded();
+            label = Guard.NotNullOrEmptyOrWhitespace(nameof(label), label, true);
+
+            var hdkey = Global.WalletService.KeyManager
+                .GenerateNewKey(new SmartLabel(label), KeyState.Clean, isInternal: false);
+            return new
+            {
+                address = hdkey.GetP2wpkhAddress(Global.Network).ToString(),
+                keyPath = hdkey.FullKeyPath.ToString(),
+                label = hdkey.Label,
+                publicKey = hdkey.PubKey.ToHex(),
+                p2wpkh = hdkey.P2wpkhScript.ToHex()
+            };
+        }
+
+        [JsonRpcMethod("getstatus")]
+        public object GetStatus()
+        {
+            var sync = Global.Synchronizer;
+
+            return new
+            {
+                torStatus = sync.TorStatus == TorStatus.NotRunning ? "Not running" : (sync.TorStatus == TorStatus.Running ? "Running" : "Turned off"),
```

Comment:
These are verbose like answers. I would provide something like a true/false interface too. Or is this common in RPC?

Reply:
There are three possible values ("Running", "Not Running" and "Turned off") so, it could be 0, 1 and 2 but then you, as a rpc consumer, should now what those numbers mean. For example: in bitcoin core rpc, the call `getchaintips` returns the following values for the field `status`:             `invalid`, `headers-only`, `valid`, `valid-headers`, `valid-fork`, and `active` as strings.

The same is valid not only for the returned values but also for the receive values in the rpc request, for example, in the `sendtoaddress` the values for fee target are the strings             "       `UNSET`, `ECONOMICAL` and `CONSERVATIVE`.

So, yes, it is common in rpc to return/receive strings. However, now that you mention it, it could be possible to improve a bit this by replacing the current strings as follow: `running, not-running, turned-off`. What do you say?